### PR TITLE
apply strict in_array with pass true to 3rd parameter

### PR DESCRIPTION
**Pull request by samsonasik**
Ensure to make strict `in_array`.

**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

File: system/Validation/CreditCardRules.php [hunk 1]
```diff
@@ -245,9 +245,11 @@ public function valid_cc_number(string $ccNumber = null, string $type): bool
 		}
 
 		// Make sure it's a valid length for this card
-		$lengths = explode(',', $info['length']);
+		$lengths = array_map(function ($value) {
```
Comment (MGatner) [inline system/Validation/CreditCardRules.php:248]: Is this one going to be worth the extra effort to ensure strict types?

Comment (samsonasik) [reply system/Validation/CreditCardRules.php:248]: phpstan complain about that, I may add `@phpstan-ignore-line` ignore it if you want.

Comment (samsonasik) [reply system/Validation/CreditCardRules.php:248]: It seems casting to int is required:

```php
$info = [
    'length' => '1,2,3',  
];

$lengths = explode(',', $info['length']);
var_dump(in_array(1, $lengths, true));
var_dump(in_array('1', $lengths, true));

$lengths = array_map(function ($value) {
    return (int) $value;
}, explode(',', $info['length']));
var_dump(in_array(1, $lengths, true));
```

Got the following output:

```
bool(false)
bool(true)
bool(true)
```

see https://3v4l.org/lPubnY

Comment (samsonasik) [reply system/Validation/CreditCardRules.php:248]: another solution is cast the `strlen()` itself to string.

```php
if (! in_array((string) strlen($ccNumber), $lengths, true))
```

Comment (samsonasik) [reply system/Validation/CreditCardRules.php]: Updated to cast the `strlen()` value itself to string. It should be ok now.

File: system/Images/Handlers/BaseHandler.php [hunk 2]
```diff
@@ -361,12 +361,12 @@ public function rotate(float $angle)
 	{
 		// Allowed rotation values
 		$degs = [
-			90,
-			180,
-			270,
+			(float) 90,
+			(float) 180,
+			(float) 270,
```
Comment (paulbalandan) [inline system/Images/Handlers/BaseHandler.php:366]: Why not just write these as `90.0`, `180.0`, `270.0`?

Comment (samsonasik) [reply system/Images/Handlers/BaseHandler.php:366]: because we can pass "90" as float ref https://3v4l.org/BFb9u

Comment (MGatner) [reply system/Images/Handlers/BaseHandler.php:366]: I don't understand the difference. Isn't `(float) 90 === 90.0`?

File: system/Validation/CreditCardRules.php (hunk 1 above)
Comment (MGatner) [reply system/Validation/CreditCardRules.php]: I meant if the strict version of `in_array` was worth the trouble of a custom array mapping, for performance and code clarity.

File: system/Images/Handlers/BaseHandler.php (hunk 2 above)
Comment (samsonasik) [reply system/Images/Handlers/BaseHandler.php:366]: oh, you're correct.  I will update to `90.0`, not in front of pc right now

Comment (samsonasik) [reply system/Images/Handlers/BaseHandler.php]: fixed.

**Comment by samsonasik** [comment]
All green :tada:

**Pull request merged**